### PR TITLE
Update res test dim

### PR DESCRIPTION
**Pull request by rmyers808**

### CODE REVIEW

File: plugins/outputs/orionres/processor/resultdef.go [hunk 1]
```diff
@@ -14,13 +14,15 @@ type ResultDefDim struct {
...
+	Enabled             bool
+	RemapResFacts       bool
+	TestDim             bool
+	PrimaryDimensionSet string
+	Dims                []*ResultDefDim
+	DimAttribs          map[string]string
+	ResFacts            map[string]*info.ResSetFact
 }
 
 func newResultDef(defName string, setDefs *info.MetricDefs) *ResultDef {
```
Comment (jhiga) [inline plugins/outputs/orionres/processor/resultdef.go:28]: below, if `defName` is not found, is it okay if `PrimaryDimensionSet` is empty string?  I assume that `TestDim` should be false so no need to explicitly set it as such.

File: plugins/outputs/orionres/res/session/session.go [hunk 2]
```diff
@@ -107,7 +107,17 @@ func (r *sessionResource) Delete(req *http.Request, id string) (int, interface{}
 }
 
 func (r *sessionResource) Update(req *http.Request, id string, value interface{}) (int, interface{}) {
-	return http.StatusNotFound, nil
+	s := value.(*Session)
```
Comment (jhiga) [inline plugins/outputs/orionres/res/session/session.go]: As written, a panic will occur if the type assertion fails.  Suggest returning `http.StatusBadRequest` if the type assertion fails.

**Review by jhiga — CHANGES_REQUESTED** [review]

File: plugins/outputs/orionres/processor/resultdef.go (hunk 1 above)
Comment (rmyers808) [reply plugins/outputs/orionres/processor/resultdef.go:28]: Good point.  I think I will default to TestDim=true, and PrimaryDimensionSet="test".  This allows setting the configured test if set and doesn't seem to cause a problem if test filed left empty.

File: plugins/outputs/orionres/res/session/session.go (hunk 2 above)
Comment (rmyers808) [reply plugins/outputs/orionres/res/session/session.go]: Fixed to return bad request and updated object

**Review by jhiga — APPROVED** [review]

**Pull request merged**